snowpack/src/build/process.ts

```diff
@@ -110,8 +110,6 @@ export async function createBuildState(commandOptions: CommandOptions): Promise<
   const isSSR = !!config.buildOptions.ssr;
   const isHMR = getIsHmrEnabled(config);
 
-  // Seems like maybe we shouldn't be doing this...
-  config.buildOptions.resolveProxyImports = !config.optimize?.bundle;
```

Comment:
What’s the reason for this change? This is a big change to bundled behavior.

Reply:
Oh—sorry. Just saw the linked issue / PR. Let me take a look at those.

Reply:
See comment here: https://github.com/snowpackjs/snowpack/pull/3194#issuecomment-825875933

I’d like for this codechange **NOT** to land, because it‘s a breaking change. Reason our tests don’t catch it is because… well we just don’t have enough optimization tests. This is much riskier than our current test suite would indicate.

Reply:
Cool no worries, I was just writing a test for the change. Maybe we should look at increasing the coverage of the optimization tests generally in order to mitigate the chance of breakage in the future.

Reply:
Yeah I’d love to see that! And yeah as you pointed out in the PR, a file appearing that wasn’t there before was a hint that the output was very different. But in general, we have too few optimization tests. I added one recently using your new test setup and it wasn’t that hard to add (`cssModules`)! Once we have more of those, we can see how PRs like #3194 affect the output a lot better.

Reply:
Indeed! Alright well I will close this here then and can look at adding more optimization tests once I've wrapped my head around what it is supposed to do in all the various configurations.